src/Twig/Component/WishlistDetails/WishlistItemsComponent.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\WishlistPlugin\Twig\Component\WishlistDetails;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Sylius\Bundle\OrderBundle\Factory\AddToCartCommandFactoryInterface;
+use Sylius\Bundle\UiBundle\Twig\Component\TemplatePropTrait;
+use Sylius\Component\Core\Factory\CartItemFactoryInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Order\Context\CartContextInterface;
+use Sylius\Component\Order\Modifier\OrderItemQuantityModifierInterface;
+use Sylius\TwigHooks\LiveComponent\HookableLiveComponentTrait;
+use Sylius\WishlistPlugin\Command\Wishlist\AddSelectedProductsToCart;
+use Sylius\WishlistPlugin\Command\Wishlist\RemoveSelectedProductsFromWishlist;
+use Sylius\WishlistPlugin\Command\Wishlist\WishlistItem;
+use Sylius\WishlistPlugin\Entity\WishlistInterface;
+use Sylius\WishlistPlugin\Repository\WishlistRepositoryInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\Attribute\LiveAction;
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\LiveComponent\ComponentToolsTrait;
+use Symfony\UX\LiveComponent\DefaultActionTrait;
+
+#[AsLiveComponent]
+final class WishlistItemsComponent
+{
+    use ComponentToolsTrait;
+    use DefaultActionTrait;
+    use TemplatePropTrait;
+    use HookableLiveComponentTrait;
+
+    /** @var array<int, bool> */
+    #[LiveProp(writable: true)]
+    public array $selection = [];
+
+    #[LiveProp]
+    public int $wishlistId;
+
+    #[LiveProp]
+    public int $itemsCount = 0;
+
+    public function __construct(
+        private readonly WishlistRepositoryInterface $wishlistRepository,
+        private readonly CartContextInterface $cartContext,
+        private readonly AddToCartCommandFactoryInterface $addToCartCommandFactory,
+        private readonly CartItemFactoryInterface $cartItemFactory,
+        private readonly OrderItemQuantityModifierInterface $orderItemQuantityModifier,
+        private readonly MessageBusInterface $messageBus,
+        private readonly RequestStack $requestStack,
+        private readonly UrlGeneratorInterface $urlGenerator,
+        private readonly TranslatorInterface $translator,
+    ) {
+    }
+
+    #[LiveAction]
+    public function toggleAll(): void
+    {
+        $all = [];
+        for ($i = 0; $i < $this->itemsCount; ++$i) {
+            $all[$i] = !($this->selection[$i] ?? false);
+        }
+        $this->selection = $all;
+    }
```

Comment:
Shouldn't this either select/deselect all? ATM, it seems to flip the current selection